server/src/main/java/org/opensearch/cluster/routing/allocation/RebalanceConstraints.java

```diff
@@ -27,9 +29,13 @@ public class RebalanceConstraints {
 
     private Map<String, Constraint> constraints;
 
-    public RebalanceConstraints() {
+    public RebalanceConstraints(float preferPrimaryBalanceBuffer) {
```

Comment:
Should we model the argument as RebalanceParameters or something similar so that we dont bombard the args list when more constraints get added in future.